--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,7 +28,6 @@
 find_package(Fcitx5ModuleEmoji REQUIRED)
 find_package(Pthread REQUIRED)
 find_package(Gettext REQUIRED)
-find_package(X11 REQUIRED)
 
 include("${FCITX_INSTALL_CMAKECONFIG_DIR}/Fcitx5Utils/Fcitx5CompilerSettings.cmake")
 add_definitions(-DFCITX_GETTEXT_DOMAIN=\"fcitx5-lotus\") 
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -8,7 +8,6 @@
   Fcitx5::Module::Emoji
 	Bamboo::Core
 	Pthread::Pthread
-	X11::X11
 )
 
 target_include_directories(lotus PRIVATE
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -16,7 +16,6 @@
     Fcitx5::Module::Emoji
     Bamboo::Core
     Pthread::Pthread
-    X11::X11
 )
 
 # Helper macro to compile standalone headless integration test linked with Lotus engine.
@@ -41,7 +40,6 @@
         Fcitx5::Module::Emoji
         Bamboo::Core
         Pthread::Pthread
-        X11::X11
     )
     add_test(NAME ${target} COMMAND ${target})
     set_tests_properties(${target} PROPERTIES LABELS "${labels}" TIMEOUT 20)
